importexport/musicxml/exportxml.cpp

```diff
@@ -4330,13 +4330,39 @@ void ExportMusicXml::ottava(Ottava const* const ot, int staff, const Fraction& t
 
 void ExportMusicXml::pedal(Pedal const* const pd, int staff, const Fraction& tick)
       {
+      // "change" type is handled only on the beginning of pedal lines
+      if (pd->tick() != tick && pd->endHookType() == HookType::HOOK_45)
+            return;
       directionTag(_xml, _attr, pd);
       _xml.stag("direction-type");
+      QString pedalType;
       QString pedalXml;
-      if (pd->tick() == tick)
-            pedalXml = "pedal type=\"start\" line=\"yes\"";
-      else
-            pedalXml = "pedal type=\"stop\" line=\"yes\"";
+      if (pd->tick() == tick) {
+            switch (pd->beginHookType()) {
+                  case HookType::HOOK_45:
+                        pedalType = "change";
+                        break;
+                  case HookType::NONE:
+                        if (pd->beginText() == "") {
+                              pedalType = "resume";
+                              break;
+                              }
```

Comment:
Is this supposed to fall though in the else case?

Reply:
Yes; a pedal with a NONE HookType but with beginText != "" is of pedalType "start"

Reply:
OK, then a `// FALLTHROUGH` is in order, to prevent a compiler warning and make the intention clear to human readers too

Reply:
#8481